[jit] Made a list for element-wise ops.

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59581 [nnc] Added micro-benchmark to show perf improvement with cat subgraph optimization
* #59580 [nnc]  Added a transformation to move users of aten::cat to its inputs in the NNC kernel subgraph.
* #60162 [jit] Exported a method to get the supported list of elementwise ops
* **#59579 [jit] Made a list for element-wise ops.**

Differential Revision: [D28955319](https://our.internmc.facebook.com/intern/diff/D28955319)

navahgar committed:
[jit] Made a list for element-wise ops.

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28955319).

eellison reviewed (approved):
LGTM

navahgar committed:
Update on "[jit] Made a list for element-wise ops."


Differential Revision: [D28955319](https://our.internmc.facebook.com/intern/diff/D28955319)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28955319).

bertmaher commented on torch/csrc/jit/passes/tensorexpr_fuser.cpp:
```diff
@@ -164,21 +164,32 @@ static const OperatorSet& supported_eltwise_set() {
-      "aten::matmul(Tensor self, Tensor other) -> Tensor",
+      // NOTE: Do not add any non-element-wise ops in this list. This set
+      // should be exclusively for element-wise ops. There are some
+      // optimizations whose correctness depend on this invariant.
```
I guess I shouldn't complain about comments but it feels a little redundant to have a note telling people not to add something that clearly doesn't belong based on the set name.

Yes conv2d and matmul shouldn't have been there (batch norm is actually debatable, it's eltwise in inference), but I'm not sure we need to carry forward the comment.

I guess I don't *really* care though, I'm just opining for no great reason :-p

navahgar replied:
I don't have a strong opinion either way. I just wanted to avoid the case of something like `matmul` getting added here again. Now that there is another set for `non-elementwise` ops, may be this wouldn't happen again.

navahgar replied:
Removed this comment.

navahgar committed:
Update on "[jit] Made a list for element-wise ops."


Differential Revision: [D28955319](https://our.internmc.facebook.com/intern/diff/D28955319)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28955319).

navahgar committed:
Update on "[jit] Made a list for element-wise ops."


Differential Revision: [D28955319](https://our.internmc.facebook.com/intern/diff/D28955319)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28955319).

navahgar committed:
Update on "[jit] Made a list for element-wise ops."


Differential Revision: [D28955319](https://our.internmc.facebook.com/intern/diff/D28955319)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28955319).
